### PR TITLE
build: update `rest-app` module for spotbug 4.3.0 and re-enable plugin

### PR DESCRIPTION
**Pull request by mjsax**
Follow up to #7808

### CODE REVIEW

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/Command.java [hunk 1]
```diff
@@ -109,10 +112,18 @@ public String getStatement() {
   }
 
   @JsonProperty("streamsProperties")
+  @SuppressFBWarnings(
+      value = "EI_EXPOSE_REP",
+      justification = "overwriteProperties is unmodifiableMap()"
+  )
   public Map<String, Object> getOverwriteProperties() {
-    return Collections.unmodifiableMap(overwriteProperties);
```
Comment (mjsax) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/Command.java:113]: Is already an `unmodifiableMap()`, no need to double-wrap.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/LagReportingResource.java [hunk 2]
```diff
@@ -15,15 +15,17 @@
...
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 import io.confluent.ksql.rest.EndpointResponse;
 import io.confluent.ksql.rest.entity.LagReportingMessage;
 import io.confluent.ksql.rest.entity.LagReportingResponse;
 import io.confluent.ksql.rest.server.LagReportingAgent;
 
 public class LagReportingResource {
 
-  private LagReportingAgent lagReportingAgent;
+  private final LagReportingAgent lagReportingAgent;
```
Comment (mjsax) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/LagReportingResource.java:26]: Side improvement.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/StatusResource.java [hunk 3]
```diff
@@ -40,10 +42,8 @@ public EndpointResponse getStatus(final String type, final String entity, final
 
     final Optional<CommandStatus> commandStatus = statementExecutor.getStatus(commandId);
 
-    if (!commandStatus.isPresent()) {
-      return Errors.notFound("Command not found");
-    }
+    return commandStatus.map(EndpointResponse::ok)
+        .orElseGet(() -> Errors.notFound("Command not found"));
 
-    return EndpointResponse.ok(commandStatus.get());
```
Comment (mjsax) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/StatusResource.java:47]: Side cleanup.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResource.java [hunk 4]
```diff
@@ -421,9 +423,9 @@ private EndpointResponse handlePullQuery(
           true
       );
       resultForMetrics.set(result);
-      result.onCompletionOrException((v, t) -> {
-        optionalDecrementer.ifPresent(Decrementer::decrementAtMostOnce);
-      });
+      result.onCompletionOrException(
+          (v, t) -> optionalDecrementer.ifPresent(Decrementer::decrementAtMostOnce)
+      );
```
Comment (mjsax) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResource.java:428]: Side cleanup.

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/api/TestEndpoints.java [hunk 5]
```diff
@@ -57,7 +60,7 @@
   private JsonObject lastProperties;
   private JsonObject lastSessionVariables;
   private String lastTarget;
-  private Set<TestQueryPublisher> queryPublishers = new HashSet<>();
+  private final Set<TestQueryPublisher> queryPublishers = new HashSet<>();
```
Comment (mjsax) [inline ksqldb-rest-app/src/test/java/io/confluent/ksql/api/TestEndpoints.java:63]: Side improvement.

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/SystemAuthenticationFunctionalTest.java [hunk 6]
```diff
@@ -50,7 +50,6 @@
 import io.confluent.ksql.util.PageViewDataProvider;
 import io.vertx.core.net.SocketAddress;
 import java.io.IOException;
-import java.nio.file.Files;
```
Comment (mjsax) [inline ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/SystemAuthenticationFunctionalTest.java:53]: Side cleanup (unused). Same below.

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/TemporaryEngine.java [hunk 7]
```diff
@@ -118,7 +119,6 @@ protected void after() {
     serviceContext.close();
   }
 
-  @SuppressWarnings("unchecked")
```
Comment (mjsax) [inline ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/TemporaryEngine.java:121]: Side cleanup. Unnecessary.

**Review by agavra — APPROVED** [review]

**Pull request merged**